hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/metrics/TestMetricsThrottleExceptions.java

```diff
@@ -0,0 +1,319 @@
+    assertTrue("All threads should complete within timeout", completed);
+    assertEquals("No exceptions should occur during concurrent access", 0, exceptions.get());
+
+    // Verify the final counter value
+    verifyCounter(testRegistry,
+      "RpcThrottlingException_Type_NumRequestsExceeded_User_alice_Table_users",
+      numThreads * incrementsPerThread);
+
+    executor.shutdown();
+  }
+
+  @Test
+  public void testCommonTableNamePatterns() {
+    // Create a test registry for the metrics
+    MetricRegistryInfo registryInfo = getRegistryInfo();
+    MetricRegistry testRegistry = MetricRegistries.global().create(registryInfo);
+    MetricsThrottleExceptions throttleMetrics = new MetricsThrottleExceptions(testRegistry);
+
+    // Test common HBase table name patterns that should be preserved
+    throttleMetrics.recordThrottleException(RpcThrottlingException.Type.NumRequestsExceeded,
+      "service-user", "my-app-logs");
+    throttleMetrics.recordThrottleException(RpcThrottlingException.Type.WriteSizeExceeded,
+      "batch.process", "namespace:table-name");
+    throttleMetrics.recordThrottleException(RpcThrottlingException.Type.ReadSizeExceeded,
+      "user_123", "test_table_v2");
+
+    // Verify common patterns are preserved correctly (note: colon gets replaced with underscore)
+    verifyCounter(testRegistry,
+      "RpcThrottlingException_Type_NumRequestsExceeded_User_service-user_Table_my-app-logs", 1);
+    verifyCounter(testRegistry,
+      "RpcThrottlingException_Type_WriteSizeExceeded_User_batch.process_Table_namespace_table-name",
+      1);
+    verifyCounter(testRegistry,
+      "RpcThrottlingException_Type_ReadSizeExceeded_User_user_123_Table_test_table_v2", 1);
+  }
+
+  @Test
+  public void testAllThrottleExceptionTypes() {
+    // Create a test registry for the metrics
+    MetricRegistryInfo registryInfo = getRegistryInfo();
+    MetricRegistry testRegistry = MetricRegistries.global().create(registryInfo);
+    MetricsThrottleExceptions throttleMetrics = new MetricsThrottleExceptions(testRegistry);
+
+    // Test all 13 throttle exception types from RpcThrottlingException.Type enum
+    RpcThrottlingException.Type[] throttleTypes = { RpcThrottlingException.Type.NumRequestsExceeded,
+      RpcThrottlingException.Type.RequestSizeExceeded,
+      RpcThrottlingException.Type.NumReadRequestsExceeded,
+      RpcThrottlingException.Type.NumWriteRequestsExceeded,
+      RpcThrottlingException.Type.WriteSizeExceeded, RpcThrottlingException.Type.ReadSizeExceeded,
+      RpcThrottlingException.Type.RequestCapacityUnitExceeded,
+      RpcThrottlingException.Type.ReadCapacityUnitExceeded,
+      RpcThrottlingException.Type.WriteCapacityUnitExceeded,
+      RpcThrottlingException.Type.AtomicRequestNumberExceeded,
+      RpcThrottlingException.Type.AtomicReadSizeExceeded,
+      RpcThrottlingException.Type.AtomicWriteSizeExceeded,
+      RpcThrottlingException.Type.RequestHandlerUsageTimeExceeded };
```

Comment:
What if `RpcThrottlingException` adds another enum type?  This unit test wont cover the new type unless it's added here.  Similarly, removing a type will break the test.  You can use something like this do dynamically get the enum types?
```suggestion
RpcThrottlingException.Type[] throttleTypes = RpcThrottlingException.Type.values();
```